### PR TITLE
feat: Bring Chakra-ui theme API v1 to Vue 2

### PR DESCRIPTION
**Pull request by simpletrontdip**
Support **Chakra UI Theme API v1** in **Vue 2**
- Use global CSS variables
- Pseudo selectors
- Base styles in themes
- Better RTL support, because of logical style attributes

## Description
- Utilize `@chakra-ui/styled-system` to bring new Theme API to Vue 2
- Remove `config` dir, now they are supported by the upstream theme package.
- Add a new serializer `@emotion/jest/serializer` to enhance snapshot tests
- Remove some assertions in tests in favor of the new serializer
- Refine the importing logic, use `@chakra-ui/vue-theme` instead of the relative path `lib/theme/src`
- Small nit for removing debug code, storybook

## Motivation and Context
- Theme **API v1** provides a better way to styling component
- This makes the theme consistent across frameworks
- We do have another repo for this update, but it is for Vue3 only
- It is nicer to have the new features for the existing lib, Vue, not everyone can update and wait for Vue3

## How Has This Been Tested?
- All existing tests passed
- New tests added
- Existing Storybook works just fine
- Most of the changes is:
  - Snapshot change, previously they are just random class names, now they include all CSS details
  - Code utils removing, these utils are now provided by upstream packages

## Screenshots:

**Update**
Sandbox demo for baseStyle [here](https://codesandbox.io/s/chakra-ui-vue-0x-starter-template-forked-6ds5x?file=/src/main.js)
```jsx
Vue.use(Chakra, {
  extendTheme: {
    baseStyle: {
      CButton: {
        borderWidth: "4px",
        borderColor: "blue.600",
        borderRadius: "lg"
      },
      CHeading: {
        fontWeight: 600,
        textDecor: "line-through"
      }
    }
  }
});
```
Screen Shot 2021-07-20 at 13 35 55

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. 
       **Not really sure about this, should we link to upstream theme API [page](https://chakra-ui.com/docs/theming/theme)**
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by codebender828** [comment]
Very interesting PR @simpletrontdip 😀  Thanks for taking the initiative on this. since it's relatively big, I will take some time to review it thoroughly. But appreciate the hard work! Will keep you posted here.

File: packages/chakra-ui-core/src/utils/components.js [hunk 1]
```diff
@@ -70,24 +69,32 @@ export const createStyledAttrsMixin = (name, isPseudo) => ({
...
+      const { nativeAttrs } = this.splitProps
+      const styleObjectOrFn = __get(this.theme, `baseStyle.${name}`)
+      return (
+        runIfFn(styleObjectOrFn, {
+          theme: this.theme,
+          colorMode: this.colorMode,
+          ...nativeAttrs
+        }) || {}
+      )
+    },
```
Comment (codebender828) [inline packages/chakra-ui-core/src/utils/components.js:82]: Very good work here 👍🏽

**Review by codebender828 — APPROVED** [review]
Thank you for the work done here, @simpletrontdip 👍🏽

Really impressed. Will be sure to give you a shoutout. I'm curious to see how you and your team are using this library in production. 😀

Will schedule a new minor version release with this change after merging it. If you'd like to continue contributing to Chakra UI Vue, I'd be happy to chat with you since we need more people to help us as we grow!

Cheers, and thanks for the good work.

**Comment by simpletrontdip** [comment]
Thanks, @codebender828.
After some evaluations, my team has decided to use Vue3 because we are starting a new project and utilize the composition API.
Not sure I have a consistent time slot for this, but I'm ready to help, just find me in the discord server, with the same id.
For Vue 3 version, we've created a JS version to speed up the development and prioritize our requirements, hope that we can contribute back someday.

**Pull request merged**